src/store/modules/util/actions.ts

```diff
@@ -98,6 +100,40 @@ const actions: ActionTree<UtilState, RootState> = {
 
     commit(types.UTIL_LOCATION_TYPES_UPDATED, locationTypes)
   },
+  async fetchRoles({ commit, state }) {
+    if (state.roles.length) {
+      return
+    }
+
+    const roles = {} as any
+    const params = {
+      inputFields: {
+        roleTypeGroupId: 'FACILITY_PARTY_ROLE'
+      },
+      viewSize: 100,
+      entityName: 'RoleTypeGroupMemberAndRoleType',
+      orderBy: 'sequenceNum',
+      filterByDate: 'Y',
+      fieldList: ['roleTypeId', 'description']
+    }
+
+    try {
+      const resp = await UtilService.fetchRoles(params)
+      if (!hasError(resp)) {
+        resp.data.docs.map((doc:any) => {
+          roles[doc.roleTypeId] = doc.description
+        })
+
+        // pushing none explicitly to show on UI
+        roles[''] = 'none'
+      } else {
+        throw resp.data
+      }
+    } catch (error) {
+      logger.error(error)
+    }
+    commit(types.UTIL_ROLES_UPDATED, roles)
```

Comment:
Improve state name, mutation to something like `facilityPartyRoles`, `UTIL_FACILITY_PARTY_ROLES_UPDATED`